File: Codeforces/1. ICPC Assiut University Community/Sheet #1 (Data type - Conditions)/Z. Hard Compare/main.cpp
// https://codeforces.com/group/MWSDmqGsZm/contest/219158/problem/Z

#include <iostream>
#include <cmath>
using namespace std;

int main()
{
    int a, c;
    long long b, d;
    cin >> a >> b >> c >> d;

    double logAB = b*log(a);
    double logCD = d*log(c);
    if(logAB > logCD){
        cout << "YES\n";
    }else{
        cout << "NO\n";
    }
}

File: Codeforces/1. ICPC Assiut University Community/Sheet #1 (Data type - Conditions)/Z. Hard Compare/main.cs
// https://codeforces.com/group/MWSDmqGsZm/contest/219158/problem/Z

using System;

namespace Codeforces.Assuit.Sheet1
{
    class ProblemZ
    {
        static void Main(string[] args)
        {
            string[] input = Console.ReadLine().Split(" ");

            int a = int.Parse(input[0]);
            long b = long.Parse(input[1]);
            int c = int.Parse(input[2]);
            long d = long.Parse(input[3]);

            double logAB = b * Math.Log(a);
            double logCD = d * Math.Log(c);

            Console.WriteLine((logAB > logCD? "YES" : "NO"));
        }
    }
}
